libmozevent/utils.py

```diff
@@ -140,9 +140,10 @@ class AsyncRedis(object):
     """
 
     async def __aenter__(self):
-        self.conn = await aioredis.create_redis(os.environ["REDIS_URL"])
-        return self.conn
+        self.redis = await aioredis.from_url(
```

Comment:
> We shouldn't cache the connection that we create but instead use a connection pool. If we cache the connection then we are unable to mitigate the risk of loosing the connection.

According to the aioredis docs, the connection pool has autoreconnecting. So we could actually cache the connection to the connection pool and rely on aioredis to reconnect to Redis when the connection is lost.

Reply:
wdyt if we make conn static like AsyncRedis.conn and set it to None and the first time when AsyncRedis is called we do AsyncRedis.conn = await aioredis.from_url()

Reply:
Yeah, that should work!

Reply:
In this case we no longer need `__aexit__`.

Reply:
Yeah...So maybe it doesn't make sense to make it a context manager anymore and just turn it into a normal class:
```
class AsyncRedis(object):
    pool = None

    @staticmethod
    async def connect():
        if AsyncRedis.pool is None:
            AsyncRedis.pool = await ...
        return AsyncRedis.pool
```

User:
```
redis = await AsyncRedis.connect
```